### ISSUE TITLE
File handle leak that causes windows agents to not cleanup properly

### ISSUE THREAD

**Issue by ThomasSkelgaard** [gocd/gocd#1743]
After upgrading to 15.3 the test artifacts from working dir on agent is locker after stage is finish. The next stage will then fail as it is not able to clean working directory.

Sometime the files are unlocked after a long time or the Go Agent needs to be re-started to free the filelock.

Is this a known issue?

Stage that complete but seems to lock file
![gostagethatlock](https://cloud.githubusercontent.com/assets/6838005/11975183/644638c6-a96b-11e5-873b-9d287f4bc916.jpg)

The stage right after that cannot do a clean
![go](https://cloud.githubusercontent.com/assets/6838005/11975137/0d942f74-a96b-11e5-8dc9-38e407d38f47.jpg)

**Comment by ketan** [comment]
@ThomasSkelgaard - we seem to think this issue is fixed by #1744. We are treating this as high priority enough to roll out a 15.3.1 release.

**Comment by jyotisingh** [comment]
Was able to reproduce it on 15.3.0 with the below config and a single agent running both stages:
``` xml
<pipelines group="defaultGroup">
    <pipeline name="tesy">
      <materials>
        <git url="C:\test3" />
      </materials>
      <stage name="defaultStage">
        <jobs>
          <job name="defaultJob">
            <tasks>
              <exec command="copy">
                <arg>c:\a.xml</arg>
                <arg>a.xml</arg>
                <runif status="passed" />
              </exec>
              <exec command="ls">
                <arg>-l</arg>
                <runif status="passed" />
              </exec>
            </tasks>
            <artifacts>
              <test src="a.xml" />
            </artifacts>
          </job>
        </jobs>
      </stage>
      <stage name="second">
        <jobs>
          <job name="job">
            <tasks>
              <fetchartifact pipeline="" stage="defaultStage" job="defaultJob" srcfile="a.xml">
                <runif status="passed" />
              </fetchartifact>
              <exec command="ls">
                <arg>-l</arg>
                <runif status="passed" />
              </exec>
            </tasks>
          </job>
        </jobs>
      </stage>
    </pipeline>
  </pipelines>
```
The second stage fails with the following error while cleaning up git repo:
```
05:57:59.101 [GIT] Cleaning all unversioned files in working copy
05:57:59.226 Error performing command: --- Command ---
git clean -dff
--- Environment ---
{}
--- INPUT ----


--OUTPUT ---
Removing cruise-output/
--- ERROR ---
STDERR: warning: failed to remove a.xml
---

05:57:59.507 [go] Job completed tesy/2/second/1/job on jyotiwindows2 [C:\GoAgent]

```

Verified that it works as expected after the fix is applied.
